parmed/gromacs/gromacstop.py

```diff
@@ -1422,6 +1423,19 @@ def write(self, dest, combine=None, parameters='inline'):
                               atom_type.mass, atom_type.charge, atom_type.sigma/10,
                               atom_type.epsilon*econv))
             parfile.write('\n')
+            # Nonbonded parameters
+            if self.has_NBFIX():
+                typemap = dict(self.parameterset.nbfix_types)
+                dest.write('[ nonbond_params ]\n')
+                for key, val in typemap.items():
+                    a1 = key[0]
+                    a2 = key[1]
+                    eps = val[0] # kcal
+                    sig = val[1] # Angstrom
+                    eps *= u.kilocalorie.conversion_factor_to(u.kilojoule)
+                    sig *= 0.1
+                    dest.write('{0} {1} 1 {2} {3}\n'.format(
```

Comment:
What is the 1 here for? Is it always 1?

Reply:
That is `nbfunc`, non-bonded function type. It is 1 for Lennard-Jones or 2 for Buckingham. From what I can tell, ParmEd will read in 2 and set it with `self.nbfunc` if given such a top file, but in this case I'm using `from_structure` and I don't think there is a corresponding attribute to in `Structure`. I can change it to this, which should handle reading and writing a `2` as long as it's always a `GromacsTopologyFile`, but this would need to be set to 1 if converted from a `Structure`.

```
                    dest.write('{0} {1} {2} {3} {4}\n'.format(
                        key[0], key[1], self.nbfunc, sig/2**(1/6), eps))
```